content/articles/Integrating-YouTube-videos-into-a-Django-app/index.md

```diff
@@ -0,0 +1,309 @@
+			</button>
+		<a  class="navbar-brand"  href="#">BLOG</a>
+		<div  class="collapse navbar-collapse"  id="navbarTogglerDemo03">
+		<ul  class="navbar-nav me-auto mb-2 mb-lg-0">
+			<li  class="nav-item">
+			<a  class="nav-link active"  aria-current="page"  href="http://127.0.0.1:8000/">Home</a>
+		</ul>
+		</div>
+	</div>
+	</nav>
+
+	{% block page_content %}{%endblock%}
+
+	</div>
+
+	<!-- Option 1: Bootstrap Bundle with Popper -->
+
+	<script  src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.1/dist/js/bootstrap.bundle.min.js"  integrity="sha384-/bQdsTh/da6pkI1MST/rWKFNjaCP5gBSY4sEBT38Q/9RBh9AH40zEOg7Hlq2THRZ"  crossorigin="anonymous"></script>
+
+</body>
+</html>
+```
+
+In `blog.html` we will display all the tutorials that will be added, but we will only display the titles and a small part of the tutorial body in a list fashion.
+
+```HTML
+{% extends 'base.html' %}
+
+	{% block page_content %}
+
+	<div  class="col-md-8 offset-md-2">
+		<h1>Blog Index</h1>
+		<hr>
+		{% for t in Tut %}
+		<h2><a  href="{% url 'blog_detail' t.pk %}">{{ t.tutorial_Title}}</a></h2>
+		<p>{{ t.tutorial_Body | slice:':400' }}...</p>
+		{% endfor %}
+	</div>
+
+	{%endblock%}
+```
+
+In `blogdetail.html` we will display the title, body and video of the selected tutorial.
+
+```HTML
+{% extends 'base.html' %} {% load embed_video_tags %}
+{% block page_content %}
+
+	<div  class="col-md-8 offset-md-2">
+		<h1>{{ Tut.tutorial_Title }}</h1>
+		<p>{{ Tut.tutorial_Body | linebreaks }}</p>
+		<h3>Here is the video:</h3>
+
+		{% video Tut.tutorial_Video 'tiny' %}
+
+	</div>
+
+{%endblock%}
+```
+
+The tag `{% video Tut.tutorial_Video 'tiny' %}` is responsible for rendering our youTube video.
+The videos can be rendered in different sizes as follows: tiny (420x315), small (480x360), medium (640x480), large (960x720) and huge (1280x960) ie: {% video Tut.tutorial_Video 'huge' %} can be used for huge videos.
+We can also set custom dimensions for the video if the default ones don't fit our requirements ie:
+{% video Tut.tutorial_Video '600x400' %}
+
+### Step 4: Testing
+
+Now that we have set up everything, we can run our app, we can run the server
+
+```bash
+py manage.py runserver
+```
+
+then login into your [admin site](http://127.0.0.1:8000/admin/) and add a few dummy tutorials to your site for testing purposes.
+
+![my admin site after adding a few tutorials](/engineering-education/Integrating-YouTube-videos-into-a-Django-app/admin.jpg)
+
+After adding the tutorials you can check out the [blog](http://127.0.0.1:8000)
+
+![my homepage after adding some tutorials](/engineering-education/Integrating-YouTube-videos-into-a-Django-app/blog.jpg)
+
+Here is how my dummy tutorials look like.
+
+![my dummy tutorial](/engineering-education/Integrating-YouTube-videos-into-a-Django-app/blogdetail.jpg)
+
+### Futher reading
+
+For more, you can check out the [documentation](https://django-embed-video.readthedocs.io/en/latest/) for django embed video.
+### Conclusion
```

Comment:
For your conclusion, add a few lines to explain why using Django or adding YouTube videos is important.

Reply:
> For your conclusion, add a few lines to explain why using Django or adding YouTube videos is important.

Am afraid if I do this, I might be repeating what I said in the introduction.

Reply:
Let me know what you think.